float32 advanceLeft
float32 advanceRight
